ui/src/timeMachine/components/SubmitQueryButton.tsx

```diff
@@ -26,7 +26,7 @@ interface StateProps {
 }
 
 interface DispatchProps {
-  onSubmit: typeof saveAndExecuteQueries
+  onSubmit: typeof saveAndExecuteQueries | (() => void)
```

Comment:
this temporary until `submit` is implemented?  if so, add a TODO?

Reply:
no, this is because saveAndExecuteQueries _is_ `() => void`, but carries the context of the redux store along for the ride, specifically that redux is going to have a collection of queries ready in the store to be executed when this button is pressed. We're reusing this component, but without redux, and it a different context, so that's why this is being extended. same interface, just without the implicitness